extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -795,39 +795,47 @@ export class AmpStoryPage extends AMP.BaseElement {
       mediaEl.play();
       return Promise.resolve();
     } else {
-      return mediaPool
-        .play(/** @type {!./media-pool.DomElementDef} */ (mediaEl))
-        .catch(unusedError => {
-          if (!this.isMediaDisplayed_(mediaEl)) {
-            return;
-          }
-
-          // Auto playing the media failed, which could be caused by a data
-          // saver, or a battery saving mode. Display a message so we can
-          // get a user gesture to bless the media elements, and play them.
-          if (mediaEl.tagName === 'VIDEO') {
-            this.debounceToggleLoadingSpinner_(false);
-
-            // If the error came from the <video> and not from a <source>
-            // descendant.
-            if (mediaEl.error) {
-              this.toggleErrorMessage_(true);
-              return;
-            }
+      if (!this.isMediaDisplayed_(mediaEl)) {
+        return Promise.resolve();
+      }
 
-            // If autoplay got rejected, display a "play" button. If autoplay
-            // was supported, dispay an error message.
-            this.isAutoplaySupported_().then(isAutoplaySupported => {
-              isAutoplaySupported
-                ? this.toggleErrorMessage_(true)
-                : this.togglePlayMessage_(true);
+      return this.loadPromise(mediaEl).then(
+        () => {
+          return mediaPool
+            .play(/** @type {!./media-pool.DomElementDef} */ (mediaEl))
+            .catch(unusedError => {
+              // Auto playing the media failed, which could be caused by a data
+              // saver, or a battery saving mode. Display a message so we can
+              // get a user gesture to bless the media elements, and play them.
+              if (mediaEl.tagName === 'VIDEO') {
+                this.debounceToggleLoadingSpinner_(false);
+
+                // If the error came from the <video> and not from a <source>
+                // descendant.
+                if (mediaEl.error) {
```

Comment:
Doesn't this also need to check `mediaEl[MEDIA_LOAD_FAILURE_SRC_PROPERTY]`?

Reply:
Yes, thanks! This test has been moved in the `loadPromise` and it'll trigger the error callback, so this (we're in the loadpromise success callback here) can be removed :)